app/domain/authentication/authn_jwt/validate_status.rb

```diff
@@ -0,0 +1,123 @@
+        validate_role_can_access_webservice: ::Authentication::Security::ValidateRoleCanAccessWebservice.new,
+        validate_webservice_exists: ::Authentication::Security::ValidateWebserviceExists.new,
+        enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str,
+        validate_account_exists: ::Authentication::Security::ValidateAccountExists.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authenticator_status_input]
+    ) do
+      extend(Forwardable)
+      def_delegators(:@authenticator_status_input, :authenticator_name, :account,
+                     :username, :status_webservice, :service_id, :client_ip)
+
+      def call
+        @logger.info(LogMessages::Authentication::AuthnJwt::ValidatingJwtStatusConfiguration.new)
+        validate_generic_status_validations
+        validate_secrets
+        @logger.info(LogMessages::Authentication::AuthnJwt::ValidatedJwtStatusConfiguration.new)
+      end
+
+      private
+
+      def validate_generic_status_validations
+        validate_account_exists
+        validate_service_id_exists
+        validate_user_has_access_to_status_webservice
+        validate_authenticator_webservice_exists
+        validate_webservice_is_whitelisted
+      end
+
+      def validate_account_exists
+        @validate_account_exists.(
+          account: account
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedAccountExists.new)
+      end
+
+      def validate_service_id_exists
+        raise Errors::Authentication::AuthnJwt::ServiceIdMissing unless service_id
+
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedServiceIdExists.new)
+      end
+
+      def validate_user_has_access_to_status_webservice
+        @validate_role_can_access_webservice.(
+          webservice: status_webservice,
+          account: account,
+          user_id: username,
+          privilege: 'read'
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedUserHasAccessToStatusWebservice.new)
+      end
+
+      def validate_authenticator_webservice_exists
+        @validate_webservice_exists.(
+          webservice: webservice,
+          account: account
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedAuthenticatorWebServiceExists.new)
+      end
+
+      def validate_webservice_is_whitelisted
+        @validate_webservice_is_whitelisted.(
+          webservice: webservice,
+          account: account,
+          enabled_authenticators: @enabled_authenticators
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedStatusWebserviceIsWhitelisted.new)
+      end
+
+      def validate_secrets
+        validate_signing_key_secrets
+        validate_issuer
+        validate_identity_secrets
+      end
+
+      def validate_signing_key_secrets
+        @create_signing_key.call(authentication_parameters: authentication_parameters)
```

Comment:
how is the validating anything? will it raise an error if the signing key secret is not there?

Reply:
also - don't you want a debug message as the rest of the class?

Reply:
yes it will raise an error
 and about the debug message,  `CreateSigningKeyFactory` command class has a log in the start of it